packages/node/src/types.ts

```diff
@@ -6,7 +6,18 @@ import type { NodeTransportOptions } from './transports';
 
 export interface EsmLoaderHookOptions {
   include?: Array<string | RegExp>;
-  exclude?: Array<string | RegExp>;
+  exclude?: Array<string | RegExp> /**
+   * When set to `true`, `import-in-the-middle` will only wrap ESM modules that are specifically instrumented by
+   * OpenTelemetry plugins. This is useful to avoid issues where `import-in-the-middle` is not compatible with some of
+   * your dependencies.
+   *
+   * **Note**: This feature will only work if you `Sentry.init()` the SDK before the instrumented modules are loaded.
+   * This can be achieved via the Node `--import` CLI flag or by loading your app via async `import()` after calling
+   * `Sentry.init()`.
+   *
+   * Defaults to `false`.
+   */;
+  onlyIncludeInstrumentedModules?: boolean;
```

Comment:
m: Are we sure we want to put this option in here? The API feels a bit "clunky" to me 🤔 also, does this not conflict with include/exclude options, somehow (or do they play together?)?

What about making this a top-level option? Esp. if we think this may become a default at some point...? (no strong feelings, just want us to consider this!)

Reply:
> does this not conflict with include/exclude options, somehow (or do they play together?)?

include/exclude were always mutually exclusive because It doesn't make much sense to use both. You could potentially enable this new `onlyIncludeInstrumentedModules ` option while still supplying `include` and this would only include instrumented modules, plus whatever the user supplied. This might be useful if a user is adding an instrumentation later.

> Are we sure we want to put this option in here? What about making this a top-level option? Esp. if we think this may become a default at some point...?

Yep valid points. I don't feel that strongly either way.